### DISCUSSION TITLE
between vmware and host machine , mitmproxy doesnot show traffic

### DISCUSSION THREAD

**Discussion by m-ersinfb13** [mitmproxy/mitmproxy#6650] [Q&A] [answered]
i have a web site which is dvwa with xampp in virtual machine. also mitmproxy is running in virtual machine. In vm i can connect dvwa website by writing http://localhost/dvwa and mitmproxy also shows the web traffic. I can also access dvwa website writing http://vm_ip_address/dvwa from my host machine. but i couldnot see web traffic in mitmproxy(which is in virtual machine).
could you please help me how can i see traffic with mitmproxy in vm?

both vm and host machine os is windows.

**Comment by emanuele-em** [comment]
Hi! I think this is exactly what you need 😃:
[https://docs.mitmproxy.org/stable/howto-transparent-vms/](https://docs.mitmproxy.org/stable/howto-transparent-vms/)

**Reply by m-ersinfb13** [reply]
i will try but i consider about it will work or not. i want to explain more about my problem: 
i have only one host machine and  inside of this host machine has only one vm.
i will use my host machine(not vm) to attack dvwa which is in vm. i want to sniff web traffic from host machine to exactly running dvwa and mitmproxy same vm.
psi: i am not sure that windows 10 can change iptables which you said in your solution.

**Comment by m-ersinfb13** [comment, marked as answer]
i solved my problem. setting vmware as bridged network and at host machine i did proxy setting as ip adress= virtual_machine_ip and port number= mitmproxy_port_number.. when you sniff , just adjust 0.0.0.0 for mitmproxy that listen all interface.
at the host machine , write url to virtual_machine_ip:port_number/dvwa
thank you for your answer also.

**Reply by emanuele-em** [reply]
Nice, thanks for sharing your solution
